File: packages/intercom/ios/Plugin/Classes/Options/LogEventOptions.swift
import Foundation
import Capacitor

@objc public class LogEventOptions: NSObject {
    let data: [String: Any]?
    let name: String

    init(_ call: CAPPluginCall) throws {
        guard let name = call.getString("name") else {
            throw CustomError.nameMissing
        }
        self.name = name
        self.data = call.getObject("data")
    }
}

File: packages/intercom/android/src/main/java/io/capawesome/capacitorjs/plugins/intercom/classes/options/LogEventOptions.java
package io.capawesome.capacitorjs.plugins.intercom.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.intercom.classes.CustomExceptions;

public class LogEventOptions {

    @Nullable
    private final JSObject data;

    @NonNull
    private final String name;

    public LogEventOptions(@NonNull PluginCall call) throws Exception {
        String name = call.getString("name");
        if (name == null) {
            throw CustomExceptions.NAME_MISSING;
        }
        this.name = name;
        this.data = call.getObject("data");
    }

    @Nullable
    public JSObject getData() {
        return data;
    }

    @NonNull
    public String getName() {
        return name;
    }
}
